src/youtube-player/youtube-player.ts

```diff
@@ -117,6 +150,16 @@ export class YouTubePlayer implements AfterViewInit, OnDestroy {
   }
 }
 
+/** Listens to changes to the given width and height and sets it on the player. */
+function bindSizeToPlayer(
+  playerObs: Observable<YT.Player | undefined>,
+  widthObs: Observable<number>,
+  heightObs: Observable<number>
+) {
+  return combineLatest(playerObs, widthObs, heightObs)
+      .subscribe(([player, width, height]) => player && player.setSize(width, height));
```

Comment:
Any reason not to just call `setSize` in `ngOnChanges` without the streams?

Reply:
Width and height are needed for the constructor of the player as well, and I think a stream fits better when managing the player initialization. I could use ngOnChanges for the setSize call, but that won't eliminate the streams entirely, so I think it'll just add complexity if we have two different means of monitoring changes on the same value.